JASP-Desktop/widgets/boundqmlcomponentslist.cpp

```diff
@@ -78,20 +100,68 @@ void BoundQMLComponentsList::bindTo(Option *option)
 Option* BoundQMLComponentsList::createOption()
 {
 	Options* templote = new Options();
-	templote->add(_optionKeyName, new OptionVariable());
+
+	if (_termsModel->areTermsInteractions())
+		templote->add(_optionKeyName, new OptionTerm());
+	else
+		templote->add(_optionKeyName, new OptionVariable());
 	addRowComponentsDefaultOptions(templote);
 
 	OptionsTable* result = new OptionsTable(templote);
 	std::vector<Options*> allOptions;
 
-	Terms initTerms = _termsModel->getSourceTerms();
-	for (const Term& term : initTerms)
+	if (hasSource())
+	{
+		Terms initTerms = _termsModel->getSourceTerms();
+		for (const Term& term : initTerms)
+		{
+			Options* row = dynamic_cast<Options*>(templote->clone());
+			if (_termsModel->areTermsInteractions())
+			{
+				OptionTerm* optionVar = new OptionTerm();
+				optionVar->setValue(term.scomponents());
+				row->add(_optionKeyName, optionVar);
+			}
+			else
+			{
+				OptionVariables* optionVar = new OptionVariable();
+				optionVar->setValue(term.asString());
+				row->add(_optionKeyName, optionVar);
+			}
+			allOptions.push_back(row);
+		}
+	}
+	else
 	{
-		Options* row = dynamic_cast<Options*>(templote->clone());
-		OptionVariable* optionVar = new OptionVariable();
-		optionVar->setValue(term.asString());
-		row->add(_optionKeyName, optionVar);
-		allOptions.push_back(row);
+		QVariant defaultValuesVar = getItemProperty("defaultValues");
+		QList<QVariant> defaultValues = defaultValuesVar.toList();
+
+		for (const QVariant& defaultValue : defaultValues)
+		{
+			Options* row = dynamic_cast<Options*>(templote->clone());
+			OptionVariables* optionVar = new OptionVariable();
+			optionVar->setValue(_incrementCounter().toStdString());
+			row->add(_optionKeyName, optionVar);
+
+			QVariantMap defaultValueMap = defaultValue.toMap();
+			QMapIterator<QString, QVariant> it(defaultValueMap);
+			while (it.hasNext())
```

Comment:
better I think: `for(QMapIterator<QString, QVariant> it(defaultValueMap); it.hasNext(); it.next())`

Notice that this changes the `it.next()` from the beginning of the loop to the end of it. But I think that is better?
At least I assume that when you instantiate an iterator for qt it also starts at `begin()` and thus at a usable value.
While your current code seems to skip the first value...

Reply:
Here I just followed QMap documentation on how to loop on a map.

Reply:
Sure, but why not just use a for-loop instead of this weird while construction?
And are you absolutely sure you are not skipping the first value?